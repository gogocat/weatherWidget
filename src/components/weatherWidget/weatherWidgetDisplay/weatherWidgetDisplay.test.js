import React from 'react';
import { shallow } from 'enzyme';
import WeatherWidgetDisplay from './weatherWidgetDisplay';

const props = {
    widgetTitle: 'widget title',
    weatherIcon: 'someIcon',
    weatherMain: '',
    temperature: 0,
    temperatureType: 'C',
    displayWindInfo: true,
    city: 'Sydney',
    wind: 'mock wind'
};

it('renders match snapshot', () => {
    const component = shallow(<WeatherWidgetDisplay
        widgetTitle={props.widgetTitle}
        weatherIcon={props.weatherIcon}
        weatherMain={props.weatherMain}
        temperature={props.temperature}
        temperatureType={props.temperatureType}
        displayWindInfo={props.displayWindInfo}
        city={props.city}
        wind={props.wind}
    />);
    expect(component.getElements()).toMatchSnapshot();
});
