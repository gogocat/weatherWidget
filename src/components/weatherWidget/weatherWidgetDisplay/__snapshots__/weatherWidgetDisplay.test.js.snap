// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders match snapshot 1`] = `
Array [
  <div
    className="weather-widget-display"
  >
    <div
      className="weather-widget-display__card"
    >
      <h3
        className="weather-widget-display__title"
      >
        widget title
      </h3>
      <div
        className="weather-widget-display__info-container"
      >
        <img
          alt=""
          src="https://openweathermap.org/img/w/someIcon.png"
        />
        <div
          className="weather-widget-display__info"
        >
          <span>
            Sydney
          </span>
          <h1
            className="weather-widget-display__temperature"
          >
            0
            <sup
              dangerouslySetInnerHTML={
                Object {
                  "__html": "&#176;",
                }
              }
            />
          </h1>
          <small>
            <strong>
              Wind
            </strong>
             
            mock wind
            km/h
          </small>
        </div>
      </div>
    </div>
  </div>,
]
`;
